src/number/format.js

```diff
@@ -68,6 +68,29 @@ return function( number, properties ) {
 		number *= 1000;
 	}
 
+	var compactPattern, compactDigits, compactProperties, divisor, pluralForm, rounded;
+	if ( compactMap ) {
+		var zeroes = Array(Math.floor(number).toString().length).join( "0" );
+		if ( zeroes.length >= 3 ) {
+			// use default plural form to perform initial decimal shift
+			compactPattern = compactMap[ "1" + zeroes + "-count-other" ];
+			compactDigits = compactPattern.split( "0" ).length - 1;
+			divisor = zeroes.length - ( compactDigits - 1 );
+			number = number / Math.pow( 10, divisor );
+
+			// calculate rounded value to determine plural form and extract correct pattern
+			rounded = parseInt(numberFormatIntegerFractionDigits( number, minimumIntegerDigits,
+				minimumFractionDigits, maximumFractionDigits, round, roundIncrement ));
```

Comment:
You can drop this line and move the below code after `numberFormatSignificantDigits` and `numberFormatIntegerFractionDigits` calls, like this:

```diff
diff --git a/src/number/format.js b/src/number/format.js
index 85ed8fbe..5ca2cda2 100644
--- a/src/number/format.js
+++ b/src/number/format.js
@@ -68,26 +68,15 @@ return function( number, properties, pluralGenerator ) {
 		number *= 1000;
 	}
 
-	var compactPattern, compactDigits, compactProperties, divisor, pluralForm, rounded;
+	var compactPattern, compactDigits, compactProperties, divisor, pluralForm, zeroes;
 	if ( compactMap ) {
-		var zeroes = Array(Math.floor(number).toString().length).join( "0" );
+		zeroes = Array(Math.floor(number).toString().length).join( "0" );
 		if ( zeroes.length >= 3 ) {
 			// use default plural form to perform initial decimal shift
 			compactPattern = compactMap[ "1" + zeroes + "-count-other" ];
 			compactDigits = compactPattern.split( "0" ).length - 1;
 			divisor = zeroes.length - ( compactDigits - 1 );
 			number = number / Math.pow( 10, divisor );
-
-			// calculate rounded value to determine plural form and extract correct pattern
-			rounded = parseInt(numberFormatIntegerFractionDigits( number, minimumIntegerDigits,
-				minimumFractionDigits, maximumFractionDigits, round, roundIncrement ));
-			pluralForm = pluralGenerator ? pluralGenerator(rounded) : "other";
-			compactPattern = compactMap[ "1" + zeroes + "-count-" + pluralForm ] || compactPattern;
-			compactProperties = compactPattern.match( numberPatternRe );
-
-			// update prefix/suffix with compact prefix/suffix
-			prefix += compactProperties[ 1 ];
-			suffix = compactProperties[ 11 ] + suffix;
 		}
 	}
 
@@ -102,6 +91,16 @@ return function( number, properties, pluralGenerator ) {
 			minimumFractionDigits, maximumFractionDigits, round, roundIncrement );
 	}
 
+	if ( compactMap && zeroes.length >= 3 ) {
+		pluralForm = pluralGenerator ? pluralGenerator(+number) : "other";
+		compactPattern = compactMap[ "1" + zeroes + "-count-" + pluralForm ] || compactPattern;
+		compactProperties = compactPattern.match( numberPatternRe );
+
+		// update prefix/suffix with compact prefix/suffix
+		prefix += compactProperties[ 1 ];
+		suffix = compactProperties[ 11 ] + suffix;
+	}
+
 	// Remove the possible number minus sign
 	number = number.replace( /^-/, "" );
```